components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImpl.java

```diff
@@ -1153,4 +1285,107 @@ public void deleteSubscriptionPolicy(String policyName) throws APIMgtDAOExceptio
+            throws SQLException {
+        final String query = "UPDATE AM_API_POLICY SET NAME = ?, DISPLAY_NAME = ?, DESCRIPTION = ?, "
+                + "DEFAULT_QUOTA_TYPE = ?, DEFAULT_QUOTA = ?, DEFAULT_UNIT_TIME = ?, DEFAULT_TIME_UNIT = ?, "
+                + "APPLICABLE_LEVEL = ? WHERE UUID = ?";
+
+        try (PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, name);
+            statement.setString(2, displayName);
+            statement.setString(3, description);
+            statement.setString(4, quotaType);
+            statement.setInt(5, quota);
+            statement.setLong(6, unitTime);
+            statement.setString(7, timeUnit);
+            statement.setString(8, applicableLevel);
+            statement.setString(9, uuid);
+
+            statement.execute();
+        }
+    }
+
+    private static void updateApplicationPolicy(String uuid, Connection connection, String name, String displayName,
+            String description, String quotaType, int quota, String quotaUnit, int unitTime, String timeUnit)
+            throws SQLException {
+        final String query = "UPDATE AM_APPLICATION_POLICY SET NAME = ?, DISPLAY_NAME = ?, DESCRIPTION = ?, "
+                + "QUOTA_TYPE = ?, QUOTA = ?, QUOTA_UNIT = ?, UNIT_TIME = ?, TIME_UNIT = ? WHERE UUID = ?";
+
+        try (PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, name);
+            statement.setString(2, displayName);
+            statement.setString(3, description);
+            statement.setString(4, quotaType);
+            statement.setInt(5, quota);
+            statement.setString(6, quotaUnit);
+            statement.setInt(7, unitTime);
+            statement.setString(8, timeUnit);
+            statement.setString(9, uuid);
+
+            statement.execute();
+        }
+    }
+
+    private static void updateSubscriptionPolicy(String uuid, Connection connection, String name, String displayName,
+            String description, String quotaType, int quota, String quotaUnit, int unitTime, String timeUnit)
+            throws SQLException {
+        final String query =
+                "UPDATE AM_SUBSCRIPTION_POLICY SET NAME = ?, DISPLAY_NAME = ?, DESCRIPTION = ?, QUOTA_TYPE = ?, "
+                        + "QUOTA = ?, QUOTA_UNIT = ?, UNIT_TIME = ?, TIME_UNIT = ? WHERE UUID = ?";
+
+        try (PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, name);
+            statement.setString(2, displayName);
+            statement.setString(3, description);
+            statement.setString(4, quotaType);
+            statement.setInt(5, quota);
+            statement.setString(6, quotaUnit);
+            statement.setInt(7, unitTime);
+            statement.setString(8, timeUnit);
+            statement.setString(9, uuid);
+
+            statement.execute();
+        }
```

Comment:
Line 1378 to 1389 is used in three places, Better to have it in a private method. Otherwise adding an extra parameter will require to change it in 3 places